lfsapi/client.go

```diff
@@ -275,7 +284,7 @@ func newRequestForRetry(req *http.Request, location string) (*http.Request, erro
 
 	for key := range req.Header {
 		if key == "Authorization" {
-			if req.URL.Scheme != newReq.URL.Scheme || req.URL.Host != newReq.URL.Host {
```

Comment:
What do you think about preventing https => http redirects? That'd prevent the client from inadvertently sending creds over an insecure channel.